packages/pointing-frame/src/LionPointingFrame.js

```diff
@@ -0,0 +1,278 @@
+import { LionLitElement, html, css, SlotMixin } from '@lion/core';
```

Comment:
The pointer frame is just the arrow on a popover, right? Did you see how it was implemented in the polymer popover? It seems like a lot of code + an extra web component just for a pointer.

Reply:
Most of the code has to do with the UX specs of the pointer. It has to be pointing in the middle of the invoker at all times, so for center positioned popups that's easy. For the corner positions it is more complex. The pointer has to be a certain set distance from the edge of the content frame while still pointing to the middle of the invoker, and this complicates things quite a bit in that you need to give the frame some offset spacing.

The reason for the extra webcomponent is for people to be able to change the pointer svg and the distance from the edge as they please. Otherwise it could probably be styles-only.

![image](https://user-images.githubusercontent.com/36734656/57177042-3fee2c00-6e60-11e9-90dc-939305d846d6.png)
This image kind of shows you how that works. The pink border is the pointing frame which position is managed by the managePosition util. Hopefully that clarifies a bit why the complexity.

I will check the polymer popover though to see if I missed an opportunity to make things simpler.

Reply:
Ah right, the moving pointer is not something that was in the old implementation. Does makes things more complex indeed.

The pointing frame isn't implemented directly by people, right? It could be a utility or mixin implemented by the popover then.

Reply:
Hmm you're correct I think. It wouldn't work well by itself, it should be inside something that implement LocalOverlayController like lion-popup. 

So is this close to what you're thinking? pseudo example:
```js
import { myPointer } from './myPointer.svg.js';

class MyCustomTooltip extends PointingFrameMixin(LionTooltip) {
  constructor() {
    super();
    this.pointerSvg = myPointer;
    this.pointerDistanceFromEdge = 6px;
  }
}
```

@daKmoR  thoughts?

Reply:
Yep that's it +1

Reply:
I can't think of a use case where a mixin us really helpful... having a web-component you can extend is good enough for now I would say :hugs:

people will probably only use the pointing-frame like so
```html
<lion-popup>
  <button slot="invoker">click me</button>
  <lion-pointing-frame slot="content">
     My Content // or <my-complex-form>
  </lion-pointing-frame>
</lion-popup>
```

e.g. composition should be good enough here

only if you want to make a differently looking frame you will need to extends `ComicBubbleFrame extends LionPointingFrame`
```html
<lion-popup>
  <button slot="invoker">click me</button>
  <comic-bubble-frame slot="content">
     My Content // or <my-complex-form>
  </comic-bubble-frame>
</lion-popup>
```

Reply:
So the pointing frame is also the thing that positions the content?

Reply:
nope the frame is only the "frame" e.g. only visual stuff like the pointer and the margins...

the position is all done by "lion-popup" which is using the LocalOverlaySystem... so you can put in whatever you want... if you just want a div with a fixed background you can do so as well... the pointing-frame, however, is a little "smarter" as it displays the pointer based on which attributes get set by the Local Overlay System

Reply:
If we want to go with a Mixin we should do so now rather than later I think. 

The cool thing about the mixin is that if anyone makes a toast or context menu or any local overlay basically, they can quite easily make it have a pointer by just applying the mixin to their custom overlay. So I see use cases there definitely, and it prevents that people try to use pointing-frame on its own in contexts where it doesn't work, it prevents confusion that way.

I'm not sure how someone would make a comic bubble though in this way, I guess they would have to .. extend the mixin as a ComicBubbleMixin and apply that one instead?